app/models/appointment_form.rb

```diff
@@ -1,43 +1,28 @@
 class AppointmentForm
-  class InvalidAppointment < StandardError; end
-
   include ActiveModel::Model
 
   attr_reader :appointment_id
 
   validates :appointment_id, presence: true
 
-  def initialize(hash = {})
-    @appointment_id = hash[:appointment_id]
-
-    validate_appointment_dates!
+  def initialize(hash = {}, verifier: ActiveSupport::MessageVerifier)
+    @appointment_id = hash['appointment_id']
+    @verifier = verifier
   end
 
   def begin_date
-    split_date.first
+    decrypted_appointment.fetch('beginDate')
   end
 
   def end_date
-    split_date.last
+    decrypted_appointment.fetch('endDate')
   end
 
   private
 
-  def split_date
-    @split_date ||= @appointment_id.to_s.split('|')
-  end
-
-  def validate_appointment_dates!
-    return unless @appointment_id
-
-    raise InvalidAppointment unless split_date.size == 2
-
-    split_date.each do |date|
-      begin
-        Time.zone.parse(date)
-      rescue ArgumentError
-        raise InvalidAppointment
-      end
-    end
+  def decrypted_appointment
+    @decrypted_appointment ||= @verifier
```

Comment:
Is there some duplication here around how we initialise the verifier? What would the tests look like if we initialised it in the controller and injected an instance?